src/production/logic.py

```diff
@@ -175,46 +175,30 @@ def handle_self_typesetter_assignment(production_assignment, request):
     return typeset_task
 
 
-def handle_assigning_typesetter(production_assignment, request):
-    errors = []
+def check_posted_typesetter_files(article, copyedit_files, posted_files):
+    check = True
+    acceptable_file_list = list()
```

Comment:
Instead of the two loops it could be something like:
acceptable_files = `[chain(article.manuscript_files.all(), article.data_figurefiles.all())]`
or even better:
acceptable_files = `{chain(article.manuscript_files.all(), article.data_figurefiles.all())}`